strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/HostedRepositoryProvider.java

```diff
@@ -47,15 +58,22 @@ public String getAlias()
     public ArtifactInputStream getInputStream(String storageId,
                                               String repositoryId,
                                               String path)
-            throws IOException,
-                   NoSuchAlgorithmException,
-                   ArtifactTransportException
+            throws IOException
     {
         Repository repository = getConfiguration().getStorage(storageId).getRepository(repositoryId);
 
-        LayoutProvider layoutPtovider = getLayoutProviderRegistry().getProvider(repository.getLayout());
-        RepositoryPath repositoryPath = layoutPtovider.resolve(repository).resolve(path);
-        return (ArtifactInputStream) Files.newInputStream(repositoryPath);
+        final LayoutProvider layoutProvider = layoutProviderRegistry.getProvider(repository.getLayout());
+        final RepositoryPath artifactPath = layoutProvider.resolve(repository).resolve(path);
+
+        logger.debug(" -> Checking local cache for {} ...", artifactPath);
+        if (layoutProvider.containsPath(repository, path))
+        {
+            logger.debug("The artifact {} was found in the local cache", artifactPath);
+            return (ArtifactInputStream) Files.newInputStream(artifactPath);
+        }
+
+        logger.debug("The artifact {} as not found in the local cache", artifactPath);
```

Comment:
` as not found` typo, should be ` was not found`

Reply:
yes
fixed, but it's not mine :)